Framework/Kernel/src/Unit.cpp

```diff
@@ -644,56 +644,87 @@ void dSpacing::init() {
       }
     }
   }
+  // force the assumption that difc is positive in debug builds
+  assert(difc > 0.);
 }
 
 double dSpacing::singleToTOF(const double x) const {
   if (!isInitialized())
     throw std::runtime_error("dSpacingBase::singleToTOF called before object "
                              "has been initialized.");
-  return difa * x * x + difc * x + tzero;
+  if (difa == 0.)
+    return difc * x + tzero;
+  else
+    return difa * x * x + difc * x + tzero;
 }
 
+/**
+ * DIFA * d^2 + DIFC * d + T0 - TOF = 0
+ *
+ * Use the citardauq formula to solve quadratic in order to minimise loss of precision. citarqauq (quadratic spelled
```

Comment:
It should be citardauq rather than citarqauq here I think - not a pleasant word to spell at the best of times!